DIPs/1NNN-MHH.md

```diff
@@ -0,0 +1,242 @@
+initializer this shall be resolved to a `string[]` containing the fully qualified names of the declarations, if any, the expression's parent UDA declaration is attached to.
+
+Note that the exact name (i.e. "ATTRIBUTE") is easily changed.
+## Contents
+- [Let UDAs see what they are attached to](#)
+  - [Abstract](#abstract)
+  - [Contents](#contents)
+  - [Rationale](#rationale)
+  - [Prior Work](#prior-work)
+  - [Description](#description)
+  - [Reference](#reference)
+  - [Copyright & License](#copyright--license)
+  - [Reviews](#reviews)
+
+## Rationale
+For a UDA to do work in D, it currently must be accessed via the scope in which the declarations it is attached to are declared. For most uses this is ideal: for example when serializing a data structure using a UDA is an efficient and maintainable way of specifying the semantics of how the given data structure is to be serialized. 
+### A simple example of a UDA - Serializing an exam result
+```D
+struct Serialize {
+    string impl;
+	this(string forWhom)
+    {
+    	impl = forWhom;
+    }
+}
+struct ExamResult {
+    //The student's name is needed in all reports
+    @Serialize("ParentReport") @Serialize("InternalReport") 
+    string studentName;
+    //The examiner's name is only needed for internal moderation
+    @Serialize("InternalReport") 
+    string examinerName;
+    //Same as the first
+    @Serialize("ParentReport") @Serialize("InternalReport") 
+    float overallScore;
+    //Don't serialize this
+    string dbEntry;
+}
+```
+
+Clearly in this case there is a strictly top-down topology to the usage of these user-defined attributes, whenever this structure is serialized D's metaprogramming facilities makes it trivial to *iterate* (emphasis added for later relevance) over `ExamResult`'s members and look for any attributes relevant to our little library.
+### A less simple example
+There are, however, situations where a UDA is highly useful but there does not exist as clean of a hierarchy as in the previous example.
+
+For example, an attribute will often be attached to a - in effect - free standing symbol like a `unittest`. 
+```D
+enum runMyTest;
+
+@runMyTest unittest {
+
+}
+```
+Since most of these declarations are at or near module scope, we can simply iterate over *all* the `unittest`s looking for one's matching our UDA. Although not technically required, it is still common to find these tests in a pre-compilation step - or by manually setting the test runner to work using a `mixin` statement.
+
+This pattern is fine for `unittest`s: When used, there are usually many of them, and when there are not many the potential for a performance hit is mitigated by the compiler providing a list of `unittest`s within a given scope.
+
+### The behaviour to be enabled by this DIP.
+There are, however, situations where this pattern is less than ideal. Some UDAs may be used to declare things which are both sparse and not `unittest`s. 
```

Comment:
UDAs are *never* used to declare things. UDAs are *attatched* to delarations.

Reply:
I will change it, but I was using the word in the sense of the English language rather than the D specification, i.e. I declare my intent to do X,Y,Z